### PR TITLE
Issue#62_Search Fixes

### LINKED ISSUE #62: Search bar for networks and sites
Adding a search bar for networks and sites.

### PR DESCRIPTION
**Pull request by purukaushik**
Fixes #62.
- Dev version [here](http://dev-site-sonar.herokuapp.com/networksbyloadtime)
- Search enabled in sitesby and networksby pages. 
- Search is conducted on either networks or sites based on which site you hit search from. 
- If you search from `/sitesbyloadtime`, results are websites, and vice versa.

⚠️ Don't merge yet 👇 👇 ⚠️ 
### Known Bug : To be Fixed
- networkstats and sitestats endpoints don't work 😞 
- searching from `/search` page searches by sites. Dunno what to do here - remove search bar or give another search bar with two buttons - **Search By Network** and **Search by Sites**

### CODE REVIEW

**Comment by purukaushik** [comment]
@FrancescoSTL  r?

**Comment by FrancescoSTL** [comment]
@purukaushik networkstats and sitestats seem to be working for me. I think for the "search" page, we just have a drop down which specifies which thing they are searching for. We can auto populate that by passing param to the page whenever we reach it from networkby or sitesby pages (ie, click on search from network by load time or network by file size and it will auto search for networks).

If a user navigates directly to the search page (or navigates there from any page that doesn't specify a searchFor parameter) we can just auto set the drop down to Networks and then they can change to sites if they want.

**Comment by purukaushik** [comment]
@FrancescoSTL 
I have fixed the first issue - ie the networkstats/sitestats page redirect on search result click
Will implement the drop down for categorical search :+1:

**Comment by FrancescoSTL** [comment]
@purukaushik 🏄

**Comment by purukaushik** [comment]
Done these:
- Display "No search results" when 0 results
- Radio button option when searching from /search page
  @FrancescoSTL  r? - Probably after you fix #69

**Comment by FrancescoSTL** [comment]
Ready to rebase. Rather than a radio button, can we do a drop down? I think its the standard way of sorting search. Also added a few nit comments

File: public/css/custom.css [hunk 1]
```diff
@@ -226,3 +226,7 @@ h1 {
   display: inline-block;
   border-radius: 50%;
 }
+
+#hidden{
+  display: none
+}
```
Comment (FrancescoSTL) [inline public/css/custom.css]: nit: add line at EOF

File: routes/index.js [hunk 2]
```diff
@@ -778,7 +778,53 @@ router.get('/sitestats', function(req, res){
 
 });
 
+router.get('/search', function(req, res){
+
```
Comment (FrancescoSTL) [inline routes/index.js:781]: Nit: remove line

File: views/layout.html [hunk 3]
```diff
@@ -47,6 +69,10 @@
 
     </div> <!-- /container -->
 
+    <!-- font awesomatic of the aromale -->
+
```
Comment (FrancescoSTL) [inline views/layout.html:73]: Nit: remove line

File: views/layout.html [hunk 4]
```diff
@@ -64,7 +90,14 @@
       ga('create', 'UA-82101385-1', 'auto');
       ga('send', 'pageview');
     </script>
-
+    <!-- If page is /search, enable radio buttons-->
+    <script>
```
Comment (FrancescoSTL) [inline views/layout.html:94]: We should probably break these scripts out into separate js files.

File: routes/index.js (hunk 2 above)
Comment (purukaushik) [reply routes/index.js:781]: Done

File: public/css/custom.css (hunk 1 above)
Comment (purukaushik) [reply public/css/custom.css]: Done

**Comment by purukaushik** [comment]
@FrancescoSTL r fixes done. 
Will deploy this to [dev](http://dev-site-sonar.herokuapp.com/) and you can have a look once before merging the changes.

**Comment by purukaushik** [comment]
:warning: not done. i see issues in dev

**Comment by purukaushik** [comment]
Fixed. Ready for merge.

File: views/layout.html [hunk 5]
```diff
@@ -26,6 +26,28 @@
...
+        <div class="col-sm-3 col-md-3 pull-right">
+          <form class="navbar-form" role="search" method="get" action="/search">
+            <div class="input-group">
+              <input type="text" class="form-control" placeholder="Search" name="query" id="srch-term">
+              <div class="input-group-btn">
+                <button class="btn btn-default" type="submit"><i class="fa fa-search" aria-hidden="true"></i></button>
+              </div>
+            </div>
+            <div class="input-group" id="hidden">
+              <div class="radio">
```
Comment (FrancescoSTL) [inline views/layout.html:38]: Can you change the radio button to dropdown? It seems to be the standard way of denoting type of search

Comment (purukaushik) [reply views/layout.html:38]: well umm, this [guy](http://uxmovement.com/forms/stop-misusing-select-menus/) recommends radio buttons for selecting fewer than 5 options.
tl;dr;
screenshot_radio

**Comment by purukaushik** [comment]
@FrancescoSTL  added nit fix for spaces - check dev

**Comment by FrancescoSTL** [comment]
r+
